sdk/src/azure/iot/az_iot_provisioning_client.c

```diff
@@ -558,3 +561,51 @@ AZ_NODISCARD az_result az_iot_provisioning_client_parse_received_topic_and_paylo
 
   return AZ_OK;
 }
+
+AZ_NODISCARD az_result az_iot_provisioning_client_get_request_payload(
+    az_iot_provisioning_client const* client,
+    az_span custom_payload_property,
+    struct az_iot_provisioning_client_payload_options const* reserved,
+    char* mqtt_payload,
+    size_t mqtt_payload_size,
+    size_t* out_mqtt_payload_length)
+{
+  (void)reserved;
+
+  _az_PRECONDITION_NOT_NULL(client);
+  _az_PRECONDITION_IS_NULL(reserved);
+  _az_PRECONDITION_NOT_NULL(mqtt_payload);
+  _az_PRECONDITION(mqtt_payload_size > 0);
+
+  az_json_writer xJsonWriter;
```

Comment:
>xJsonWriter [](http://example.com/codeflow?start=17&length=11)

not sure if we follow the same naming convention in azure-sdk-for-c.

Reply:
Fixing this and also xBuffer.